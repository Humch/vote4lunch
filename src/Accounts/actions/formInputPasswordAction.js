export default password => ({
  type: 'FORM_INPUT_PASSWORD',
  password,
});
