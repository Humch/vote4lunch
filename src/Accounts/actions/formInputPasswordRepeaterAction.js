export default passwordRepeater => ({
  type: 'FORM_INPUT_PASSWORD_REPEATER',
  passwordRepeater,
});
