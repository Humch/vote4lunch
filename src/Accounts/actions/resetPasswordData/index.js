import resetPasswordData from './resetPasswordData';

export default resetPasswordData;
