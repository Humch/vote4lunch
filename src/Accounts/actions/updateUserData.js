export default (pseudo, email) => ({
  type: 'UPDATE_USER_DATA',
  pseudo,
  email,
});
