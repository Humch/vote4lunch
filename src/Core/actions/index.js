import formInputDate from './formInputDateAction/formInputDateAction';
import formInputEndDate from './formInputEndDateAction';
import onTopAlert from './onTopAlert';
import offTopAlert from './offTopAlert';

export {
  formInputDate,
  formInputEndDate,
  onTopAlert,
  offTopAlert
};
