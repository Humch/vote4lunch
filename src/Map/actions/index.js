import { fetchRestaurants } from './fetchRestaurants';
import offMapAlert from './offMapAlert';
import onMapAlert from './onMapAlert';
import resetListOfRestaurants from './resetListOfRestaurants';
import updateMapCoordinates from './updateMapCoordinates';
import updateZoomLevel from './updateZoomLevel';

export {
  fetchRestaurants,
  offMapAlert,
  onMapAlert,
  resetListOfRestaurants,
  updateMapCoordinates,
  updateZoomLevel,
};
