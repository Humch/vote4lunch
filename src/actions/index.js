export const saveVoteData = (id, date, pseudo, email, url) => ({
  type: 'SAVE_VOTE_DATA',
  id,
  date,
  pseudo,
  email,
  url
});

export const resetVoteData = () => ({
  type: 'RESET_VOTE_DATA',
});

export const updateUserData = (pseudo, email) => ({
  type: 'UPDATE_USER_DATA',
  pseudo,
  email,
});
