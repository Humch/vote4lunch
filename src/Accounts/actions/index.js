import { addUser } from './addUser/addUser';
import formInputPseudo from './formInputPseudoAction';
import formInputPassword from './formInputPasswordAction';
import formInputPasswordRepeater from './formInputPasswordRepeaterAction';
import formInputEmail from './formInputEmailAction';
import { checkPseudo } from './pseudoChecker/pseudoChecker';
import { checkEmail } from './emailChecker/emailChecker';
import updateUserData from './updateUserData';

export {
  addUser,
  formInputPseudo,
  formInputPassword,
  formInputPasswordRepeater,
  formInputEmail,
  checkPseudo,
  checkEmail,
  updateUserData,
};
