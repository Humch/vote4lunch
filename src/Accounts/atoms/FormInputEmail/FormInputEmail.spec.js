/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { FormInputEmail } from '../..';

import { formInputEmail } from '../../actions';

describe('FormInputEmail Snapshot', () => {
  const initialState = {
    userData: {
      email: 'bob@bob.com'
    },
    emailChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormInputEmail', () => {
    const renderedValue = renderer.create(<FormInputEmail store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});


describe('FormInputEmail (Shallow + passing the {store} directly)', () => {
  const initialState = {
    userData: {
      email: 'bob@bob.com'
    },
    emailChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<FormInputEmail store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('email')).toEqual(initialState.userData.email);
  });
});

describe('FormInputEmail - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {
    userData: {
      email: 'bob@bob.com'
    },
    emailChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();
    wrapper = mount(<Provider store={store}><FormInputEmail /></Provider>);
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check action on dispatching', () => {
    store.dispatch(formInputEmail('bob@bob.com'));
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_EMAIL');
    expect(action[0].email).toBe(initialState.userData.email);
  });

  it('check onChange in email input', () => {
    const input = wrapper.find('#email');
    expect(input.props().value).toBe(initialState.userData.email);

    const event = { target: { value: 'roger@roger.com' } };
    input.props().onChange(event);
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_EMAIL');
    expect(action[0].email).toBe('roger@roger.com');
  });
});
