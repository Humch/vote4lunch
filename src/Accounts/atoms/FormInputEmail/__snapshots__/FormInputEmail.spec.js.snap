// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputEmail Snapshot capturing Snapshot of FormInputEmail 1`] = `
<input
  className="form-control "
  id="email"
  name="email"
  onChange={[Function]}
  required={true}
  type="email"
  value="bob@bob.com"
/>
`;
